### ISSUE TITLE
Remove BinaryBitmap::isPureBarcode()

### ISSUE THREAD

**Issue by axxel** [nu-book/zxing-cpp#129]
@huycn, I assume you have decided to move the channel for the 'isPure' information from the `DecodeHints` class, where it is in upstream, in to the `BinaryBitmap` / Binarizer interface. I'd like to argue that that is conceptually the wrong place.

There are two individual but connected concerns:
 a) the image contains only 'clean' bits, i.e. no shadows/distortion/noise
 b) all 'non-background' pixels belong to exactly one perfectly aligned barcode

Both are typically only the case for generated images, but all 4 combinations can potentially occur with both scanned and generated data.

a) means a trivial threshold based binarizer is adequate and b) has nothing to do with the binarization, it means the detectors can use a fast route to find the bounding box of the code.

The current code is not tested by the blackbox tests and seems [broken](https://github.com/nu-book/zxing-cpp/issues/128#issuecomment-641520706), too. I'd like to deprecate this member function and unplug it from any internal bits. Objections/Input? (Also @vkrause)

I will add a `DecodeHints::isPure` (not really happy with the name, though) flag for the b)-part and a `DecodeHints::binarizer` enum for the a)-part. The latter will only have an effect through the `ReadBarcode` API.

**Comment by huycn** [comment]
I agree with you to move the flag `isPure` from `BinaryBitmap` back `DecodeHints`. I removed it from `DecodeHints` to simply this interface as I've failed to see any real-world scenario where we would have a "pure" bitmap, other than for unit test purpose (i.e. generated data only). It's clear that I was wrong as we can see with KDE apps.

Providing `DecodeHints::Binarizer` enum is also a good idea in my opinion, it will simplify a lot for "advanced usage" cases.

So, yes, I vote for both changes.
